custom_components/ramses_cc/__init__.py

```diff
@@ -264,44 +242,11 @@ def __init__(
     @property
     def available(self) -> bool:
         """Return True if the sensor is available."""
-        return getattr(self._device, self._state_attr) is not None
+        return getattr(self.rf_entity, self._state_attr) is not None
 
     @property
     def name(self) -> str:
         """Return the name of the binary_sensor/sensor."""
-        if not hasattr(self._device, "name") or not self._device.name:
-            return f"{self._device.id} {self._state_attr}"
-        return f"{self._device.name} {self._state_attr}"
-
-
-class RamsesZoneBase(RamsesEntity):  # for: climate & water_heater
-    """Base for any RAMSES RF-compatible entity (e.g. Controller, DHW, Zones)."""
-
-    _attr_precision: float = PRECISION_TENTHS
-    _attr_temperature_unit: str = UnitOfTemperature.CELSIUS
-
-    def __init__(self, broker, device) -> None:
-        """Initialize the sensor."""
-        super().__init__(broker, device)
-
-        # dont include platform/domain (climate.ramses_cc)
-        self._attr_unique_id = device.id
-
-    @property
-    def current_temperature(self) -> float | None:
-        """Return the current temperature."""
-        return self._device.temperature
-
-    @property
-    def extra_state_attributes(self) -> dict[str, Any]:
-        """Return the integration-specific state attributes."""
-        return {
-            **super().extra_state_attributes,
-            "schema": self._device.schema,
-            "params": self._device.params,
-        }
-
-    @property
-    def name(self) -> str | None:
```

Comment:
For now this is copy-pasted into a couple of places but will be removed in future when device info is added at base entity level as the entity names will then just become `None` to reflect the fact they are the primary entity for the HA device.

Reply:
OK, just beware some (climate) entities can be renamed.

Reply:
That should be fine, but will of course need testing properly once the HA devices are in place.

Reply:
I haven't checked yet what will be required to update the associated device name.

Reply:
I've had complaints that **ramses_cc** doesn't handle renaming of zones very well.

It would be great to start writing tests for this at some stage. Sigh... there is a lot to do.

Reply:
Would I be correct in guessing that the complaints are that the entity ID is not updated to reflect the new name?

Reply:
Yes.

Reply:
I believe it would be easy enough to provide a fixed entity ID as is [done for the sensors already](https://github.com/zxdavb/ramses_cc/blob/7fcf48ae23a7c5073addbfd25fd606c061ef3457/custom_components/ramses_cc/__init__.py#L259).

Reply:
I am not sure where we are with this, but I feel the `entity_id` should be the zone name.

That way, you can destroy/recreate your zone (it may get a new `zone_idx`, and thus a new `unique_id`). But if it has the same `entity_id`, then you don't have to re-write your automations.

Reply:
At the moment the entity id will be implicitly generated from the entity name. This is fine for the most part but has some issues:
- if discovery does not find the zone name before the entity is created the entity id will reflect the fallback name
- if the zone name is changed the entity id is not regenerated

So on balance it is probably better to explicitly provide an entity id, but I'll approach that separately as I didn't really want to change any behaviour in this PR.